src/Native/System.Security.Cryptography.Native/pal_x509.cpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_x509.h"
+
+extern "C" X509_NAME* DecodeX509Name(const unsigned char* buf, int32_t len)
```

Comment:
I agree it logically does belong there, but following the [Interop Guidelines](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/interop-guidelines.md):

>- Minimize unnecessary IL in assemblies.
  - Interop signatures should only be compiled into the assemblies that
    actually consume them. Having extra signatures bloats assemblies and
    makes it more difficult to do static analysis over assemblies to
    understand what they actually use. It also leads to problems when such
    static verification is used as a gate, e.g. if a store verifies that
    only certain APIs are used by apps in the store.

If I was to put the signature in Interop.X509Name.cs, then System.Net.Security would have a new method compiled in it that it doesn't use, and System.Security.Cryptography.X509Certificates would have 2 new methods that it doesn't use.
(Note: I like to keep the Interop.X.cs file 1:1 with pal_x.cpp/h, as I think this makes the most sense when trying to find where methods are declared.)